woocommerce/payment-gateway/class-sv-wc-payment-gateway.php

```diff
@@ -1944,13 +1944,11 @@ public function get_order( $order ) {
 		/* translators: Placeholders: %1$s - site title, %2$s - order number */
 		$order->description = sprintf( esc_html__( '%1$s - Order %2$s', 'woocommerce-plugin-framework' ), wp_specialchars_decode( SV_WC_Helper::get_site_name(), ENT_QUOTES ), $order->get_order_number() );
 
-		// when HPOS is enabled, we need to save the order to avoid potential errors when saving a new payment method
-		if ( SV_WC_Plugin_Compatibility::is_hpos_enabled() && ( ! is_admin() || is_ajax() ) ) {
-			$order->save();
+		// the get_order_with_unique_transaction_ref() call results in saving the order object, which we don't want to do if the order hasn't already been saved (such as when adding a payment method)
+		if ( $order->get_id() ) {
```

Comment:
The previous fix of checking if we were on the "Add Payment Method" page did not work. The problem was that it would fail if you were on the payment method page, but doing an ajax action (like Elavon does). So it would create orders during the ajax processing.